sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java

```diff
@@ -327,7 +400,23 @@ private void updateRetryAfterLimits(
       final long retryAfterMillis = parseRetryAfterOrDefault(retryAfterHeader);
       // we dont care if Date is UTC as we just add the relative seconds
       final Date date = new Date(System.currentTimeMillis() + retryAfterMillis);
-      sentryRetryAfterLimit.put(HTTP_RETRY_DEFAULT_CATEGORY, date);
+      applyRetryAfterOnlyIfLonger(DataCategory.All, date);
+    }
+  }
+
+  /**
+   * apply new timestamp for rate limiting only if its longer than the previous one
+   *
+   * @param dataCategory the DataCategory
+   * @param date the Date to be applied
+   */
+  private void applyRetryAfterOnlyIfLonger(
```

Comment:
`l`: Didn't know this part. That's a reason why I like PRs 😜. We have to update this in Cocoa.